tvb_nest/tests/test_models.py

```diff
@@ -0,0 +1,127 @@
+        self.nest_nodes_ids = nest_nodes_ids
+        self.nest_model_builder = nest_model_builder
+        self.interface_model_builder = interface_model_builder
+        self.results_path = os.path.join(os.getcwd(), "outputs",
+                                         self.interface_model_builder.__name__.split("Builder")[0])
+        self.model_params = model_params
+
+    def run(self):
+        delete_folder_safely(self.results_path)
+        return main_example(self.model, self.nest_model_builder, self.interface_model_builder,
+                            self.nest_nodes_ids, nest_populations_order=self.nest_populations_order,
+                            tvb_to_nest_mode=self.tvb_to_nest_mode, nest_to_tvb=self.nest_to_tvb,
+                            exclusive_nodes=self.exclusive_nodes, delays_flag=self.delays_flag,
+                            simulation_length=self.simulation_length, transient=self.transient,
+                            **self.model_params)
+
+
+class TestWilsonCowan(TestModel):
+    tvb_to_nest_mode = "rate"
+
+    def __init__(self):
+        super(TestWilsonCowan, self).__init__(WilsonCowan, [33, 34], WilsonCowanBuilder,
+                                              InterfaceWilsonCowanBuilder, model_params_wc)
+
+
+class TestWilsonCowanMultisynapse(TestModel):
+    tvb_to_nest_mode = "rate"
+
+    def __init__(self):
+        super(TestWilsonCowanMultisynapse, self).__init__(WilsonCowan, [33, 34], WilsonCowanMultisynapseBuilder,
+                                                          InterfaceWilsonCowanMultisynapseBuilder, model_params_wc)
+
+
+class TestReducedWongWangExcIO(TestModel):
+
+    def __init__(self):
+        super(TestReducedWongWangExcIO, self).__init__(ReducedWongWangExcIO, [33, 34], WWDeco2013Builder,
+                                                       RedWWexcIOBuilder, model_params_redww_exc_io)
+
+    def run(self):
+        for tvb_to_nest_mode in ["param", "current", "rate"]:
+            self.tvb_to_nest_mode = tvb_to_nest_mode
+            super(TestReducedWongWangExcIO, self).run()
+
+
+class TestReducedWongWangExcIOinhI(TestModel):
+
+    def __init__(self):
+        super(TestReducedWongWangExcIOinhI, self).__init__(ReducedWongWangExcIOInhI, [33, 34], WWDeco2014Builder,
+                                                           RedWWexcIOinhIBuilder, model_params_redww_exc_io_inn_i)
+
+    def run(self):
+        for tvb_to_nest_mode in ["param", "current", "rate"]:
+            self.tvb_to_nest_mode = tvb_to_nest_mode
+            super(TestReducedWongWangExcIOinhI, self).run()
+
+
+class TestIzhikevichRedWWexcIO(TestModel):
+    tvb_to_nest_mode = "rate"
+
+    def __init__(self):
+        super(TestIzhikevichRedWWexcIO, self).__init__(ReducedWongWangExcIO, list(range(10)),
+                                                       BasalGangliaIzhikevichBuilder, IzhikevichRedWWexcIOBuilder, {})
+
+
+@pytest.mark.skip(reason="These tests are taking too much time")
```

Comment:
Have you tried to run these with a length = 10ms, nest_populations_order=10 and transient=1, for example?

Reply:
This test is no longer slow. 
The problem was the simulation of a model SpikingReducedWongWang that has several modes (4th dimension) representing neurons.
I have excluded this model now, and also reduced the numbers as you are suggesting.